### PR TITLE
Improve some Haddock in BlockFetch

### PR DESCRIPTION
**Pull request by nfrisby**
Updated some stale Haddock references. Improved wording and added details to `FetchDecline` Haddock.

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs
Comment (amesgen) [inline ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:120]: By "validation", do you mean something like parsing the block CBOR in the context of BlockFetch?

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:163]: I considered it, but it'd require updating downstream code too, etc. So I thought I'd leave that for you to decide later :)

But if you're already fine with that, I could make that change in this PR.

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:120]: Thanks, I reworded this.

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:172]: Thanks. Done.

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:97]: Thanks. Done.

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:163]: Just to be clear: I'd rather not block this PR, if there's any delay in this decision.

Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:165]: That's ok, you can do that in a separate PR, or open an issue for us.

Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:165]: Thanks. I created https://github.com/input-output-hk/ouroboros-network/issues/4541

**Comment by nfrisby** [comment]
Status update: I think this PR is ready for merge, once it gets an Approval.

**Review by coot — APPROVED** [review]
LGTM

**Comment by coot** [comment]
Thanks @amesgen & @nfrisby :) 

bors merge

**Pull request merged**
